src/NSubstitute/SubstituteExtensions.cs

```diff
@@ -364,19 +368,23 @@ internal static ConfiguredCall Returns<T>(MatchArgs matchArgs, Func<CallInfo, T>
             return x => CompletedTask(returnThis(x));
         }
 
+#if !NET40
         private static Func<CallInfo, ValueTask<T>> WrapFuncInValueTask<T>(Func<CallInfo, T> returnThis)
         {
             return x => CompletedValueTask(returnThis(x));
         }
 
-        internal static Task<T> CompletedTask<T>(T result) 
+        internal static ValueTask<T> CompletedValueTask<T>(T result)
         {
-            return Task.FromResult(result);
+            return new ValueTask<T>(result);
         }
+#endif
 
-        internal static ValueTask<T> CompletedValueTask<T>(T result)
+        internal static Task<T> CompletedTask<T>(T result)
         {
```

Comment:
The GitHub diff is deceiving here - note I've just reordered these two methods for the sake of #ifdef's!